tokio/src/sync/broadcast.rs

```diff
@@ -604,13 +614,20 @@ impl<T> Sender<T> {
 
         // Slot lock acquired
         slot.write.pos.with_mut(|ptr| unsafe { *ptr = pos });
-        slot.write.val.with_mut(|ptr| unsafe { *ptr = value });
+
+        // Set the closed bit if the value is `None`; otherwise write the value
+        if value.is_none() {
+            slot.lock.fetch_or(CLOSED, SeqCst);
+            tail.closed = true;
+        } else {
+            slot.write.val.with_mut(|ptr| unsafe { *ptr = value });
+        }
 
         // Set remaining receivers
         slot.rem.store(rem, SeqCst);
 
         // Release the slot lock
-        slot.lock.store(0, SeqCst);
+        slot.lock.fetch_and(CLOSED, SeqCst);
```

Comment:
Not a huge deal, but `store` is going to be a bit lighter than `fetch_and` (store vs. read-modify-write). Because the lock is held here, you should be able to replace this w/ `store(is_closed)` if you track whether or not the  channel is being closed.

Not a blocker, so up to you.

Reply:
Good call. I moved this to where we set the closed bit or write a new value